packages/core/src/singleton/API/types.ts

```diff
@@ -1,38 +1,76 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
+
+import { Headers } from '../../clients';
+
 export type LibraryAPIOptions = {
-	AppSync: {
-		query: string;
-		variables?: object;
-		authMode?: any;
-		authToken?: string;
-		/**
-		 * @deprecated This property should not be used
-		 */
-		userAgentSuffix?: string;
+	GraphQL?: {
+		// custom headers for given GraphQL service. Will be applied to all operations.
+		headers?: () => Promise<Headers>;
+	};
+	REST?: {
+		// custom headers for given REST service. Will be applied to all operations.
+		headers?: (options: { apiName: string }) => Promise<Headers>;
 	};
-	customHeaders: Function;
 };
 
-export type APIConfig = {
-	AppSync?: {
-		defaultAuthMode: GraphQLAuthMode;
-		region: string;
-		endpoint: string;
-		modelIntrospectionSchema?: any;
-	};
+type GraphQLEndpointConfig = {
+	/**
+	 * Required GraphQL endpoint, must be a valid URL string.
+	 */
+	endpoint: string;
+	/**
+	 * Optional region string used to sign the request. Required only if the auth mode is 'iam'.
+	 */
+	region?: string;
+	/**
+	 * Optional API key string. Required only if the auth mode is 'apiKey'.
+	 */
+	apiKey?: string;
+	/**
+	 * Endpoint to custom GraphQL API.
```

Comment:
```suggestion
	 * Custom domain endpoint for GraphQL API.
```

What do you think about this?